sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -185,14 +186,18 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
   }
 
   private def isProbablyShuffleJoin(left: LogicalPlan,
-      right: LogicalPlan, hint: JoinHint): Boolean = {
-    !hintToBroadcastLeft(hint) && !hintToBroadcastRight(hint) &&
-      !canBroadcastBySize(left, conf) && !canBroadcastBySize(right, conf)
+      right: LogicalPlan,
+      hint: JoinHint,
+      joinType: JoinType): Boolean = {
+    // If any of the side can be broadcast, then it is not a shuffle join.
+    !(canLeftSideBeBroadcast(left, conf, hint, joinType) ||
+      canRightSideBeBroadcast(right, conf, hint, joinType))
```

Comment:
```
private def isProbablyShuffleJoin(
    left: LogicalPlan,
    right: LogicalPlan,
    hint: JoinHint,
    joinType: JoinType): Boolean = {
  ...
```

Reply:
@beliefer 
I have fixed the format issue. The earlier check was like this, as per @oss-maker  suggestion I have modified like this as I found it to be more intuitive.

Reply:
I think the code I mentioned is more intuitive.